scripts/build.py

```diff
@@ -14,22 +14,38 @@
 
 """Build file for production version of Oppia. Minifies JS and CSS."""
 
+import hashlib
+import json
 import os
 import re
 import shutil
 import subprocess
 import sys
-import yaml
 
-
-HEAD_DIR = os.path.join('core', 'templates', 'dev', 'head', '')
-OUT_DIR = os.path.join('core', 'templates', 'prod', 'head', '')
 REMOVE_WS = re.compile(r'\s{2,}').sub
 YUICOMPRESSOR_DIR = os.path.join(
     '..', 'oppia_tools', 'yuicompressor-2.4.8', 'yuicompressor-2.4.8.jar')
 
-FILE_EXTENSIONS_TO_IGNORE = ['.py']
-
+# Files with this extension shouldn't be moved to build folder.
+FILE_EXTENSIONS_TO_IGNORE = ('.py')
+# Files with this extension should be moved to build folder but shouldn't be
+# renamed (i.e. the filepath shouldn't contain hash)
+# This is because these files are referenced from places we can't influence.
+FILE_EXTENSIONS_NOT_TO_RENAME = (
+    '.map', '.woff', '.ttf', '.woff2', '.eot', '.ico', '.txt')
+
+# Hashes for files in these folders should be provided to frontend in
+# JS hashes object.
+FOLDERS_PROVIDED_TO_FRONTEND = (os.path.join('images', ''),
```

Comment:
What do you think about specifying both this and the next constant as a single set of glob patterns instead -- something like FILEPATHS_TO_HASH? We do something like this in scripts/pre_commit_linter.py and conceptually it might be more straightforward. Ideally we get it to the point where we can just edit that single list and all the hashing gets taken care of transparently.